crates/fluvio-protocol/src/core/wasm_bytes.rs

```diff
@@ -0,0 +1,87 @@
+use std::io::Error;
+use std::io::ErrorKind;
+
+use bytes::{Bytes, BufMut};
+
+use crate::{Encoder, Version};
+
+/// Represnts a SmartModule WASM File bytes.
+///
+/// Provides a `Encoder` implementation optimized for WASM files used in
+/// SmartModules.
+pub struct WasmBytes(Vec<u8>);
```

Comment:
Since this can be used anywhere, can we come up with more generic name?  `ByteVec`, `ByteBuf`, etc.

Reply:
`ByteBuf` sounds good to me!